consensus/mempool_test.go

```diff
@@ -166,17 +166,16 @@ func TestMempoolRmBadTx(t *testing.T) {
 		// CheckTx should not err, but the app should return a bad abci code
 		// and the tx should get removed from the pool
 		invalidTx := []byte("invalidTx")
-		err := assertMempool(cs.txNotifier).CheckTx(invalidTx, func(r *abci.ResponseCheckTx) {
-			if r.Code != kvstore.CodeTypeInvalidTxFormat {
-				t.Errorf("expected checktx to return invalid format, got %v", r)
-				return
-			}
-			checkTxRespCh <- struct{}{}
-		}, mempl.TxInfo{})
+		reqRes, err := assertMempool(cs.txNotifier).CheckTx(invalidTx)
 		if err != nil {
 			t.Errorf("error after CheckTx: %v", err)
 			return
 		}
+		if reqRes.Response.GetCheckTx().Code != kvstore.CodeTypeInvalidTxFormat {
```

Comment:
Even if the app is in-memory, I believe the verification should be async. (As in the original code base.)